table/column.go

```diff
@@ -348,7 +362,10 @@ func getColDefaultValueFromNil(ctx sessionctx.Context, col *model.ColumnInfo) (t
 	}
 	if mysql.HasAutoIncrementFlag(col.Flag) {
 		// Auto increment column doesn't has default value and we should not return error.
-		return types.Datum{}, nil
+		return GetZeroValue(col), nil
+	}
+	if col.IsGenerated() {
+		return types.NewDatum(nil), nil
```

Comment:
What if the generated column have a not null flag ?

Reply:
It will be checked in `HandleBadNull`.